stock_buffer_route/models/stock_buffer.py

```diff
@@ -35,18 +35,15 @@ def _compute_route_ids(self):
 
     def _get_location_routes_of_parents(self, routes, parents):
         return routes.filtered(
-            lambda route: any(
-                p.location_id in parents
-                for p in route.rule_ids.filtered(
-                    lambda rule: rule.action in ("pull", "pull_push", "manufacture")
-                ).mapped("location_src_id")
+            lambda route: (
+                # at least one rule of the route must have a destination location
+                # reaching the buffer
+                route.rule_ids.filtered(lambda rule: rule.action != "push").mapped(
+                    "location_id"
+                )
+                & parents
```

Comment:
to check: if the rule is `Stock/Shelving/Shelf`, and the buffer's location is `Stock/Shelving`, we are replenishing the buffer, and the reverse is not true. So I don't think we should check the parents of the buffers location, but if the rule destination location is a child of the buffer location.

Reply:
My comment is wrong, the current code matches what is done in `_get_rule`.